packages/verkle/src/constructors.ts

```diff
@@ -6,36 +6,33 @@ import { VerkleTree } from './verkleTree.js'
 
 import type { VerkleTreeOpts } from './types.js'
 
-export async function createVerkleTree(opts?: VerkleTreeOpts) {
+export async function createVerkleTree(opts?: Partial<VerkleTreeOpts>) {
   const key = ROOT_DB_KEY
 
-  if (opts?.db !== undefined && opts?.useRootPersistence === true) {
-    if (opts?.root === undefined) {
-      opts.root = await opts?.db.get(key, {
+  // Provide sensible default options
+  const parsedOptions = {
+    ...opts,
+    db: opts?.db ?? new MapDB<Uint8Array, Uint8Array>(),
+    verkleCrypto: opts?.verkleCrypto ?? (await loadVerkleCrypto()),
+    useRootPersistence: opts?.useRootPersistence ?? false,
+    cacheSize: opts?.cacheSize ?? 0,
+  }
+
+  if (parsedOptions.useRootPersistence === true) {
+    if (parsedOptions.root === undefined) {
+      parsedOptions.root = await parsedOptions.db.get(key, {
         keyEncoding: KeyEncoding.Bytes,
         valueEncoding: ValueEncoding.Bytes,
       })
     } else {
-      await opts?.db.put(key, opts.root, {
+      await parsedOptions.db.put(key, parsedOptions.root, {
         keyEncoding: KeyEncoding.Bytes,
         valueEncoding: ValueEncoding.Bytes,
       })
     }
   }
 
-  if (opts?.verkleCrypto === undefined) {
-    const verkleCrypto = await loadVerkleCrypto()
-    if (opts === undefined)
-      opts = {
-        verkleCrypto,
-        db: new MapDB<Uint8Array, Uint8Array>(),
-      }
-    else {
-      opts.verkleCrypto = verkleCrypto
-    }
-  }
-
-  const trie = new VerkleTree(opts)
-  await trie['_createRootNode']()
+  const trie = new VerkleTree(parsedOptions)
+  await trie.createRootNode()
```

Comment:
You don't want to call this if `useRootPersistence === true` and you can retrieve the `parsedOptions.root` enables you to retrieve an actual node from the db.